isisd/isis_sr.c

```diff
@@ -641,14 +641,17 @@ void sr_adj_sid_add_single(struct isis_adjacency *adj, int family, bool backup,
 			   struct list *nexthops)
 {
 	struct isis_circuit *circuit = adj->circuit;
-	struct isis_area *area = circuit->area;
+	struct isis_area *area;
 	struct sr_adjacency *sra;
 	struct isis_adj_sid *adj_sid;
 	struct isis_lan_adj_sid *ladj_sid;
 	union g_addr nexthop = {};
 	uint8_t flags;
 	mpls_label_t input_label;
 
+	area = circuit->area;
```

Comment:
There's something wrong here. There can't be an adjacency over a circuit with `circuit->area == NULL`. This change may fix your crash, but it will just hide the actual problem.